runtime/interpreter/value.go

```diff
@@ -94,7 +94,7 @@ type Value interface {
 	Accept(interpreter *Interpreter, visitor Visitor)
 	Walk(interpreter *Interpreter, walkChild func(Value))
 	DynamicType(interpreter *Interpreter, seenReferences SeenReferences) DynamicType
-	StaticType() StaticType
+	StaticType(interpreter *Interpreter) StaticType
```

Comment:
Is there a reason that `StaticType` needs to take an `*Interpreter` instead of a `MemoryGauge`. I didn't realize this until I needed to write https://github.com/onflow/cadence/pull/1558, but I think we should try to make these argument types as general as possible.

Reply:
I went with `*Interpreter` to match the other methods.

Reply:
Yeah, we should probably change the rest of those over as well.

Reply:
An interpreter is needed for the `StaticType()` method for the runtime subtype check change https://github.com/onflow/cadence/pull/1487 (if we are also including it that is). I would suggest having the interpreter only for this method because then it would cater to both requirements + fewer conflicts.
 
I agree it makes sense to have a `MemoryGauge` parameter for other methods though.